pkg/controller/elasticsearch/user/roles.go

```diff
@@ -58,6 +61,17 @@ var (
 				},
 			},
 		},
+		ApmAgentUserRole: esclient.Role{
+			Cluster: []string{},
+			Indices: []esclient.IndexRole{},
+			Applications: []esclient.ApplicationRole{
```

Comment:
@simitt @sqren would be glad to have your review on this role. Thanks 🙏

Reply:
This lgtm.
fyi: In parallel I'm working on improving this by making this privilege default for the  `apm_user` role: https://github.com/elastic/elasticsearch/issues/57201.

It might not be available until 8.0 (so still quite a long way out) but better than never.